### ISSUE TITLE
Correlate track keys between countries

### ISSUE THREAD

**Issue by steveluscher** [rdio/api#68]
Rdio is currently available in 24 countries. I've just discovered that, for the same song, a track might have key **A** in one country, but key **B** in another.

I signed up for Rdio with a Canadian credit card. If I search for, say, David Lynch's song “Pinky's Dream (trentemøller Remix)” using http://rdioconsole.appspot.com, it yields the track key “t27627990”. I assume this is its key in the US catalogue. If I authenticate using my (Canadian) account and perform the same search, the track key returned is “t19277151”. The ISRC for the two tracks is the same.

Here's the feature request:
- the option of querying the API with a track key to get corresponding track keys in other countries, identified by country code.
- the option of querying the API with a user key to get their country code.

_or_
- the option of querying the API with a user key and a track key to get the corresponding track key in that user's native catalogue, if available.

In both cases, you'd have to consider that this could leak information about what country the user's credit card is issued in, to a determined hacker. Weigh that against the benefit of being able to correlate track keys between Rdio users in different countries.

**Comment by steveluscher** [comment]
For now, I'm going to have to two-stage it. I'll have to query the catalogue for a track's ISRC, then have each client query their own country's Rdio catalogue in an attempt to turn that into a valid track key for that catalogue, if available.

**Comment by dasevilla** [comment]
You are correct, Rdio has different _releases_ for different regions because the rights holder may vary between regions. This can cause issues when you attempt to share tracks between regions.

Requesting different releases for a particular is an interesting idea, I'm not sure if we have that mapping already. We've discussed using global album IDs but actually return a region-specific release for the user's region. This would be a large change to Rdio API's data model and isn't likely to change soon. 

Until then, there are some advanced techniques you can use when you care about the user's region:
- When requesting a track or album object, you can use the `streamRegions` extras field
- You can use the `streamRegion` extras field when using the `currentUser` method

Both use the ISO two-character country codes.

**Comment by steveluscher** [comment]
I'm so glad that there's a way to check if a song is streamable in another region. Thanks!
